log/util.go

```diff
@@ -0,0 +1,57 @@
+package log
+
+import (
+	"fmt"
+	"reflect"
+)
+
+// InterleavedKVToFields converts keyValues a la Span.LogKV() to a Field slice
+// a la Span.LogFields().
+func InterleavedKVToFields(keyValues ...interface{}) ([]Field, error) {
+	if len(keyValues)%2 != 0 {
+		return nil, fmt.Errorf("Non-even keyValues len: %v", len(keyValues))
+	}
+	fields := make([]Field, len(keyValues)/2)
+	for i := 0; i*2 < len(keyValues); i++ {
+		key, ok := keyValues[i*2].(string)
+		if !ok {
+			return nil, fmt.Errorf(
+				"Non-string key (pair=%v): %v",
+				i, reflect.TypeOf(keyValues[i*2]))
+		}
+		switch typedVal := keyValues[i*2+1].(type) {
```

Comment:
What is the value of this `switch` vs. just using `Object()` to store the value as an `interface{}`? As far as I understand, the field stuff is useful to avoid the allocations involved in boxing things into `interface{}`, but that has already happened here.